### PR TITLE
Add yaml readme path field

### PR DESCRIPTION
**Pull request by SDBowen**
Added the ability to define a `readme` path in the Kitspace yaml. 

Also did some clean up of the repetitive multi repo checking by moving that logic further up the build process.

### CODE REVIEW

**Comment by kasbah** [comment]
Looks like it breaks [failing on ArduTouch](https://travis-ci.org/kitspace/kitspace/builds/629816358#L591). I noticed in the ArduTouch kitspace.yaml it says `Gerbers/output` but it should be `Gerbers/outputs` (with an `s`). Not sure how that worked before :thinking:

**Comment by kasbah** [comment]
Fixed it https://github.com/maltman23/ArduTouch/pull/7

**Comment by SDBowen** [comment]
Yeah, I noticed that I was actually not checking the yaml defined gerbers path if it was not a multi project repo.

I think this latest push will get the `readme` key working, but I noticed that the [DIY_particle_detector](https://github.com/kitspace-forks/DIY_particle_detector/) also uses html img tags in the readme; causing broken images.

So, almost there, but a little more work to do on this.

File: tasks/page/processBOM.js [hunk 1]
```diff
@@ -49,10 +49,16 @@ if (require.main !== module) {
...
-  if (file != null) {
-    kitspaceYaml = yaml.safeLoad(file)
+  try {
+    const file = fs.readFileSync(yamlPath)
+
+    if (file != null) {
+      kitspaceYaml = yaml.safeLoad(file)
+    }
+  } catch (error) {
+    console.log('WARNING: NO KITSPACE YAML FILE')
```
Comment (kasbah) [inline tasks/page/processBOM.js]: We actually allow no yaml file, if people use the default file locations.

Comment (kasbah) [reply tasks/page/processBOM.js]: I think rather than a try/catch it would be better to check if the file is valid.

**Comment by kasbah** [comment]
Looks good, clearer than before. Deleting more lines than you are adding is a good sign. I am a bit confused about the `yamlPath` in cases where there is no yaml file. I think not using try/catch for that that case will make it more explicit. 

If that's fixed I am happy to remove the test repo and merge. Fixing the images is another PR I think.

File: configure [hunk 2]
```diff
@@ -128,34 +128,47 @@ for (var f of images) {
...
-    for (let project in info.multi) {
-      projectPath = path.join(folder, project)
-      newBoardFolders.push(projectPath)
-    }
-  } else {
-    newBoardFolders.push(folder)
-  }
-  return newBoardFolders
-}, [])
+  return [{}, folder]
```
Comment (kasbah) [inline configure]: Why are you returning a folder here when there is no yaml? I can see that that could cause issues in the build system.

**Comment by ozel** [comment]
hey, owner of [DIY_particle_detector](https://github.com/ozel/DIY_particle_detector) here. Originally I wanted to feature both board layout versions as one multi. But that is a bit too confusing for my kind of users in case of upgrading from one to the other variant. Instead, two assembly variants in the sense of two different bom files and different readmes for the same set of gerbers are all I need. In those separate readmes, I won't need html img tags, I'll just link back to the main readme for further reference. So if I follow now the syntax of this [kitspace.yaml](https://github.com/kitspace-forks/DIY_particle_detector/blob/master/kitspace.yaml) and supply two different sets of readmes and boms plus the same gerbers in both cases it should work, right?

Thanks a lot, this is really great progress!

**Comment by kasbah** [comment]
Yep, I think that should work @ozel. HTML images needs to be fixed anyway, but if you stick to markdown for now then we can get it up sooner.

File: configure (hunk 2 above)
Comment (SDBowen) [reply configure]: Yeah, good catch, that was part of my issue.

Originally added it so the `deps` array would always contain four valid paths in `processBOM`; which also lead to me adding the try/catch.

Re-arranging that `deps` array, and moved the `yamlPath` to the end. Let me get rid of the try/catch and just check for the existence of `yamlPath`.

File: tasks/page/processBOM.js (hunk 1 above)
Comment (SDBowen) [reply tasks/page/processBOM.js]: Updated :+1:

**Comment by kasbah** [comment]
Needs some docs in our own readme and the test repo removed. Looks good otherwise.

**Pull request merged**